source/common/router/header_parser.cc

```diff
@@ -254,6 +254,10 @@ HeaderParserPtr HeaderParser::configure(
 
 void HeaderParser::evaluateHeaders(Http::HeaderMap& headers,
                                    const StreamInfo::StreamInfo& stream_info) const {
+  for (const auto& header : headers_to_remove_) {
+    headers.remove(header);
+  }
+
```

Comment:
I see that adding this block will solve the problem mentioned in this PR description. Can you add a comment telling the motivation of having this here?